Riot/Modules/Room/Views/BubbleCells/RoomTypingBubbleCell.swift

```diff
@@ -0,0 +1,150 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+
+@objcMembers
+class RoomTypingBubbleCell: MXKTableViewCell, Themable {
+    // MARK: - Constants
+    
+    private enum Constants {
+        static let maxPictureCount = 4
+        static let pictureSize: CGFloat = 24
+        static let pictureMaxMargin: CGFloat = 16
+        static let pictureMinMargin: CGFloat = 8
+    }
+
+    // MARK: - Outlets
+    
+    @IBOutlet private weak var additionalUsersLabel: UILabel!
+    @IBOutlet private weak var additionalUsersLabelLeadingConstraint: NSLayoutConstraint!
+    @IBOutlet private weak var dotsView: DotsView!
+    @IBOutlet private weak var dotsViewLeadingConstraint: NSLayoutConstraint!
+
+    // MARK: - members
+    
+    private var userPictureViews: [MXKImageView] = []
+    
+    // MARK: - Lifecycle
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        
+        update(theme: ThemeService.shared().theme)
+    }
+    
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        
+        for pictureView in userPictureViews {
+            pictureView.removeFromSuperview()
+        }
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+
+        dotsView.isHidden = userPictureViews.count == 0
+
+        guard userPictureViews.count > 0 else {
+            return
+        }
+
+        additionalUsersLabel?.sizeToFit()
+        
+        var pictureViewsMaxX: CGFloat = 0
+        var xOffset: CGFloat = 0
+        for pictureView in userPictureViews {
+            pictureView.center = CGPoint(x: Constants.pictureMaxMargin + xOffset + pictureView.bounds.midX, y: self.bounds.midY)
+            xOffset += round(pictureView.bounds.maxX * 2 / 3)
+            pictureViewsMaxX = pictureView.frame.maxX
+        }
+        
+        let leftMagin: CGFloat = pictureViewsMaxX + (userPictureViews.count == 1 ? Constants.pictureMaxMargin : Constants.pictureMinMargin)
+        additionalUsersLabelLeadingConstraint.constant = leftMagin
+        
+        dotsViewLeadingConstraint?.constant = additionalUsersLabel.text.isEmptyOrNil == true ? leftMagin : leftMagin + 8 + additionalUsersLabel.frame.width
+    }
+    
+    // MARK: - Overrides
+    
+    override class func defaultReuseIdentifier() -> String {
+        return String(describing: self)
+    }
+    
+    override class func nib() -> UINib {
+        return UINib(nibName: String(describing: self), bundle: nil)
+    }
+    
+    override func customizeRendering() {
+        super.customizeRendering()
+        
+        let theme = ThemeService.shared().theme
+        self.update(theme: theme)
+    }
+    
```

Comment:
```suggestion
```

Reply:
We won't need this part, as we're already updating the theme at appropriate place.